cns/service/main.go

```diff
@@ -918,11 +910,42 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 		}
 	}()
 
+	clientset, err := kubernetes.NewForConfig(kubeConfig)
```

Comment:
I still suggest to move this logic inside initCNS. It is more contained there as part of CNS initialization workflow.

Reply:
podInfoByProvider is kind of an internal state which we are passing to InitCNS func.

Reply:
moving the clientset/the rest of this initialization into the `initCNS` breaks IOC - we should provide that func all of the objects it needs as inputs. passing it ready-to-use implementations of the interfaces it needs is the better pattern